### PR TITLE
seacas: require +metis and +mpi instead of +parmetis

### PR DESCRIPTION
**Pull request by ghost**
This change aligns the build condition for parmetis with the depends_on condition.
The current build condition of parmetis looks for "+parmetis" in the spec which is not added by the depends_on as that adds "^parmetis" instead.

### CODE REVIEW

**Comment by mvlopri** [comment]
@spackbot fixstyle

**Comment by ghost** [comment]
@spackbot commands

**Comment by ghost** [comment]
@spackbot fix style

**Review by gsjaardema — APPROVED** [review]
This looks good, but do the references to parmetis in that block need to change also:
```
                    define(
                        "ParMETIS_LIBRARY_DIRS",
                        [spec["parmetis"].prefix.lib, spec["metis"].prefix.lib],
                    ),
                    define("ParMETIS_LIBRARY_NAMES", ["parmetis", "metis"]),
                    define(
                        "TPL_ParMETIS_INCLUDE_DIRS",
                        [spec["parmetis"].prefix.include, spec["metis"].prefix.include],
                    ),
```
In particular the `spec["parmetis"]`.  They might be OK, but just asking to be sure.

**Comment by sebrowne** [comment]
> In particular the `spec["parmetis"]`. They might be OK, but just asking to be sure.

`spec['parmetis']` is still fine, it gets added to the spec by means of the `depends_on('parmetis')` earlier in the file.  The issue here was the `+parmetis`, which is only true if it's a _variant_ that is then enabled.

File: var/spack/repos/builtin/packages/seacas/package.py [hunk 1]
```diff
@@ -425,7 +425,7 @@ def cmake_args(self):
             [define("TPL_ENABLE_Netcdf", True), define("NetCDF_ROOT", spec["netcdf-c"].prefix)]
         )
 
-        if "+parmetis" in spec:
+        if "+metis" in spec and "+mpi" in spec:
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/seacas/package.py]:
```suggestion
        if spec.satisfies("+metis+mpi"):
```

**Review by alalazo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**